shumway/__init__.py

```diff
@@ -86,11 +86,13 @@ def __exit__(self, *args):
 
 class MetricRelay(object):
     """Create and send metrics"""
-    def __init__(self, default_key, ffwd_ip=FFWD_IP, ffwd_port=FFWD_PORT):
+    def __init__(self, default_key, ffwd_ip=FFWD_IP, ffwd_port=FFWD_PORT,
+                 default_attributes=None):
         self._metrics = {}
         self._sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         self._default_key = default_key
         self._ffwd_address = (ffwd_ip, ffwd_port)
+        self._default_attributes = default_attributes
```

Comment:
How do you feel about taking a copy of the attributes or updating a new dict with them? It might be overkill to prevent all possible types of misuse, but I think we have bitten it in the past by treating default init parameters as constants, which doesn't hold if someone changes the dict they passed in outside the context of shumway.

Reply:
Good call, updated!